src/libraries/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.cs

```diff
@@ -43,6 +43,14 @@ public SafeSocketHandle(IntPtr preexistingHandle, bool ownsHandle)
 
         internal bool OwnsHandle { get; }
 
+        // Socket continuations are dispatched to the ThreadPool from the event thread.
+        // This avoids continuations blocking the event handling.
+        // Setting PreferInlineCompletions allows continuations to run directly on the event thread.
+        // PreferInlineCompletions defaults to false and can be set to true using the DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS envvar.
+        internal static readonly bool InlineSocketCompletionsEnabled = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS") == "1";
+
+        internal bool PreferInlineCompletions { get; set; } = InlineSocketCompletionsEnabled;
```

Comment:
I just noticed these changes are in SafeSocketHandle.cs rather than SafeSocketHandle.Unix.cs; same for the ones below in Socket.cs rather than Socket.Unix.cs.  Can they be moved to the Unix-specific files?